app/models/application_record.rb

```diff
@@ -7,14 +7,18 @@ class ApplicationRecord < ActiveRecord::Base
 
   DATA_ACCESS_PERIOD_FOR_PUBLISHERS = 1.year.freeze
 
+  def attributes_except_ciphertext
+    attributes.reject { |k, _v| k.include?("_ciphertext") }
+  end
+
   private
 
   def event_data
-    { table_name: self.class.table_name }.merge(anonymised_attributes)
+    { "table_name" => self.class.table_name }.merge(anonymised_attributes)
```

Comment:
> Why this key should change from a symbol to a string?



It's merged with attributes, which are all string keys. It's less confusing when all the keys are the same type.